test/utils/formatAmount.spec.ts

```diff
@@ -0,0 +1,118 @@
+    expect(formatAmount(new BN(toWei(new BN('1'), 'ether')))).toEqual('1')
+  })
+
+  test('12,345 Ether', async () => {
+    expect(formatAmount(new BN(toWei(new BN('12345'), 'ether')))).toEqual('12,345')
+  })
+
+  test('0100 Ether', async () => {
+    expect(formatAmount(new BN(toWei(new BN('0100'), 'ether')))).toEqual('100')
+  })
+
+  test('123,456,789,012 Ether', async () => {
+    expect(formatAmount(new BN(toWei(new BN('123456789012'), 'ether')))).toEqual('123,456,789,012')
+  })
+})
+
+describe('Exact decimal amounts', () => {
+  test('0.5 Ether', async () => {
+    expect(formatAmount(new BN(toWei(new BN('500'), 'milliether')))).toEqual('0.5')
+  })
+
+  test('1.234 Ether', async () => {
+    expect(formatAmount(new BN(toWei(new BN('1234'), 'milliether')))).toEqual('1.234')
+  })
+
+  test('1.2345 Ether', async () => {
+    expect(formatAmount(new BN(toWei(new BN('1234500'), 'microether')))).toEqual('1.2345')
+  })
+})
+
+describe('Rounding amounts', () => {
+  test('0', async () => {
+    expect(formatAmount(ZERO)).toEqual('0')
+  })
+
+  test('1 wei', async () => {
+    expect(formatAmount(ONE)).toEqual('0')
+  })
+
+  test('1000 wei', async () => {
+    expect(formatAmount(new BN('1000'))).toEqual('0')
+  })
+
+  test('1.23451 Ether, round down', async () => {
+    expect(formatAmount(new BN(toWei(new BN('1234510'), 'microether')))).toEqual('1.2345')
+  })
+
+  test('1.23456 Ether, round up', async () => {
+    expect(formatAmount(new BN(toWei(new BN('1234560'), 'microether')))).toEqual('1.2346')
+  })
+
+  test('1.234567890123456789 Ether', async () => {
+    expect(formatAmount(new BN('1234567890123456789'))).toEqual('1.2346')
+  })
+})
+
+describe('Tokens with precision 6', () => {
+  test('1 unit', async () => {
+    const amount = new BN('1000000')
+    expect(formatAmount(amount, 6)).toEqual('1')
+  })
+
+  test('12,345 units', async () => {
+    const amount = new BN('12345000000')
+    expect(formatAmount(amount, 6)).toEqual('12,345')
+  })
+
+  test('4,567,890.123444 units, round down', async () => {
+    const amount = new BN('4567890123444')
+    expect(formatAmount(amount, 6)).toEqual('4,567,890.1234')
+  })
+
+  test('4,567,890.123455 units, round up', async () => {
+    const amount = new BN('4567890123455')
+    expect(formatAmount(amount, 6)).toEqual('4,567,890.1235')
+  })
+})
+
+describe('2 decimals', () => {
+  test('1 unit', async () => {
+    const amount = new BN('1000000')
+    expect(formatAmount(amount, 6, 2)).toEqual('1')
+  })
+
+  test('12,345 units', async () => {
+    const amount = new BN('1000000')
+    expect(formatAmount(amount, 6, 2)).toEqual('1')
+  })
+
+  test('4,567,890.123333 units, round down', async () => {
+    const amount = new BN('4567890123333')
+    expect(formatAmount(amount, 6, 2)).toEqual('4,567,890.12')
+  })
+
+  test('4,567,890.125555 units, round up', async () => {
+    const amount = new BN('4567890125555')
+    expect(formatAmount(amount, 6, 2)).toEqual('4,567,890.13')
```

Comment:
Question: Should 0.5 round up to 1? 
My concept is that, from 0.0 to 0.5, round up == 0, while 0.6 to 0.9, round up == 1
Not the case?

Reply:
It's pretty common to round the `5` up
![image](https://user-images.githubusercontent.com/2352112/65529437-60458280-def6-11e9-8e9a-1f7264e5059d.png)